integration-tests/tests/src/tests/sync/app.ts

```diff
@@ -0,0 +1,322 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { ObjectId } from "bson";
+import { expect } from "chai";
+import { List } from "realm";
+import { importAppBefore } from "../../hooks";
+import { genPartition } from "../../utils/generators";
+import { TemplateReplacements } from "../../utils/import-app";
+
+const TestObjectSchema: Realm.ObjectSchema = {
+  name: "TestObject",
+  properties: {
+    doubleCol: "double",
+  },
+};
+
+const PersonForSyncSchema: Realm.ObjectSchema = {
+  name: "Person",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    age: "int",
+    dogs: "Dog[]",
+    firstName: "string",
+    lastName: "string",
+    realm_id: "string?",
+  },
+};
+
+const DogForSyncSchema: Realm.ObjectSchema = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "string?",
+  },
+};
+
+interface IPersonForSyncSchema {
+  _id: ObjectId;
+  age: number;
+  dogs: IDogForSyncSchema[];
+  firstName: string;
+  lastName: string;
+  realm_id: string | undefined;
+}
+
+interface IDogForSyncSchema {
+  _id: ObjectId;
+  breed: string | undefined;
+  name: string;
+  realm_id: string | undefined;
+}
+
+describe("App", () => {
+  describe("instantiation", function () {
+    type ConfigContext = {
+      conf: { id: string; baseUrl: string };
+    } & Mocha.Context;
+    afterEach(async () => {
+      Realm.clearTestState();
+    });
+    before(async function (this: ConfigContext) {
+      this.conf = { id: "smurf", baseUrl: "http://localhost:9090" };
+    });
+
+    it("from config", function (this: ConfigContext) {
+      const app = new Realm.App(this.conf);
+      expect(app).instanceOf(Realm.App);
+    });
+
+    it("from string", function (this: ConfigContext) {
+      const app = new Realm.App(this.conf.id);
+      expect(app).instanceOf(Realm.App);
+      expect(app.id).equals(this.conf.id);
+    });
+
+    it("throws on undefined app", function () {
+      //@ts-expect-error creating an app without a config should fail
+      expect(() => new Realm.App()).throws(Error, "Invalid arguments: 1 expected, but 0 supplied.");
+    });
+
+    it("throws on invalid input", function () {
+      //@ts-expect-error creating an app with an invalid config should fail
+      expect(() => new Realm.App(1234)).throws(Error, "Expected either a configuration object or an app id string.");
+    });
+
+    it("throws on invalid baseURL", function () {
+      const invalid_url_conf = { id: "smurf", baseUrl: "http://localhost:9999" };
+      const app = new Realm.App(invalid_url_conf);
+
+      const credentials = Realm.Credentials.anonymous();
+      console.log(credentials);
```

Comment:
We should probably remove `console.log`s statements from the tests.

Reply:
good catch, I must have mistakenly left it there while debugging